### ISSUE TITLE
3.9.21 and PHP 7.4.x compatibility issues with built in LESS compiler

### ISSUE THREAD

**Issue by Quark-Zimmerman** [joomla/joomla-cms#30569]
### Steps to reproduce the issue
1. Install 3.9.21
2. Install PHP 7.4.10
3. Install Kunena Forum 5.1.19
4. Install "Blue Eagle 5" Kunena template and make it default
5. Enable PHP error reporting 
6. Browse to Forum's index page

### Expected result
Forum displayed without problems

### Actual result
Got following errors:
Deprecated: Array and string offset access syntax with curly braces is deprecated in <site_path>\libraries\vendor\leafo\lessphp\lessc.inc.php on line 749

Deprecated: Array and string offset access syntax with curly braces is deprecated in <site_path>\libraries\vendor\leafo\lessphp\lessc.inc.php on line 1860

Deprecated: Array and string offset access syntax with curly braces is deprecated in <site_path>\libraries\vendor\leafo\lessphp\lessc.inc.php on line 2519

Deprecated: Array and string offset access syntax with curly braces is deprecated in <site_path>\libraries\vendor\leafo\lessphp\lessc.inc.php on line 2573

### Additional comments

Basically the problem is with "lessc.inc.php" that uses old/deprectaed syntax like "$subProp[1]{0}" to address arry elements instead of "$subProp[1][0]".
Not the major issue by far, but since this LESS compiler is distributed with Joomla and Joomla recommends to upgrade to 7.4 ... looks bad :)

**Comment by Quark-Zimmerman** [comment]
Attached the version of the file that fixes the issue.
Very simple , I just do not like doing core "hacking" , if this distributed with Joomla - everyone should get it as part of the install.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30569">issues.joomla.org/tracker/joomla-cms/30569</a>.</sub>

**Comment by Quark-Zimmerman** [comment]
Attached the version of the file that fixes the issue.
Very simple , I just do not like doing core "hacking" , if this distributed with Joomla - everyone should get it as part of the install.


[lessc.inc.php.txt](https://issues.joomla.org/uploads/1/5fe05908e025f61c513b37bdbcb3693b.txt)
(just remove the .txt from extension)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30569">issues.joomla.org/tracker/joomla-cms/30569</a>.</sub>

**Comment by Quy** [comment]
Duplicate #30452

**Comment by ReLater** [comment]
The issue has been "discussed" already here: https://github.com/joomla/joomla-cms/issues/30452

The only reliable solution (because it's a vendor library) would be to use another, still maintained less-php repo or provide a maintained one as replacement for the current one. BTW: As far as I know less has been removed from Joomla 4 (?).

(I used my custom system plugin in the past to load my fixed less library very early instead of the joomla less library. But that was a "very individual solution", not appropriate for Joomla core.)

**Comment by Quy** [comment]
See upstream PR https://github.com/leafo/lessphp/pull/647

**Comment by alikon** [comment]
What's the point on submitting a pr on a upstream dead project  ??
surely we should consider to fork this , if we really think we need to really support j.3  for 2 years after the release of j4 stable

**Comment by zero-24** [comment]
An topic for this issue (backporting combat fixes for dead projects) has been added to the production department meeting agenda to be held on 08.09.2020

**Comment by alikon** [comment]
well then we can simply close this

**Comment by zero-24** [comment]
Why should we close this here? Has there been an decision made? It was just me that added that to the document.

**Comment by alikon** [comment]
No,
sorry for being in hurry then let this be reopened and waiting for _Godot_

**Comment by SharkyKZ** [comment]
This errors fatally on PHP 8.

**Comment by zero-24** [comment]
To foreshadow the meeting report to be published this is an acknowledged issue where it seems we got an agreement on how we want to backport the changes. Details will follow once that thing is finalized.

**Comment by brianteeman** [comment]
As there hasnt been a published report from the department for 60 days is there any chance of an update here

**Comment by HLeithner** [comment]
We will fork the repository and fix it for j3 packages. The meeting reports will come soon because of personal things they got delayed...

**Comment by brianteeman** [comment]
Thanks for the update

**Comment by Hackwar** [comment]
This is the repo where we will do the fixes: https://github.com/joomla-projects/lessphp 
Help is appreciated.

**Comment by Quy** [comment]
@HLeithner OK to close per PR #31428?

**Comment by HLeithner** [comment]
Yes

**Comment by mattab** [comment]
fyi wikimedia maintains a fork at https://github.com/wikimedia/less.php#transitioning-from-leafolessphp

**Comment by HLeithner** [comment]
Thanks for the info @mattab we will look into it, latest versions seems to require php 7.2.9 and in case of j3 we have minimum requirement of 5.3.10. So 1.8.1 is the only option.

For j4 we don't include a less compiler.
